staging/src/kubevirt.io/client-go/api/v1/schema.go

```diff
@@ -189,7 +189,7 @@ type DomainSpec struct {
 	Memory *Memory `json:"memory,omitempty"`
 	// Machine type.
 	// +optional
-	Machine Machine `json:"machine,omitempty"`
+	Machine *Machine `json:"machine"`
```

Comment:
You need to keep `omitempty` here otherwise nil `Machine` will result in `null`  in json. This will be denied by webhook.